src/main/java/org/stellar/anchor/paymentsservice/PaymentsService.java

```diff
@@ -80,15 +80,17 @@
      * API request that executes a payment between accounts. The APIKey needs to have access to the source account for
      * this request to succeed.
      *
-     * @param sourceAccountId      the ID of the account making the payment.
-     * @param destinationAccountId the ID of the account receiving the payment.
-     * @param currencyName         the name of the currency used in the payment. It should obey the <scheme>:<identifier>
-     *                             format described in https://stellar.org/protocol/sep-38#asset-identification-format.
-     * @param amount               the payment amount.
+     * @param sourceAccount      the account making the payment. Only the network and id fields are needed.
+     * @param destinationAccount the account receiving the payment. The network field and a subset of (id, address and
+     *                           addressTag) will be mandatory.
+     * @param currencyName       the name of the currency used in the payment. It should obey the {scheme}:{identifier}
+     *                           format described in <a href="https://stellar.org/protocol/sep-38#asset-identification-format">SEP-38</a>.
+     * @param amount             the payment amount.
      * @return asynchronous stream with the payment object.
+     * @throws HttpException If the provided input parameters are invalid.
      * @throws HttpException If the http response status code is 4xx or 5xx.
      */
-    Mono<Payment> sendPayment(String sourceAccountId, String destinationAccountId, String currencyName, String amount) throws HttpException;
+    Mono<Payment> sendPayment(Account sourceAccount, Account destinationAccount, String currencyName, Float amount) throws HttpException;
```

Comment:
Amounts where precision matters are generally not handling using floating point numbers. Does Java have a BigNumber or Decimal type that is better suited for this variable?

Reply:
Looks like [BigDecimal](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html) is an option.

Reply:
Yes, BigDecimal is the right way to go.

Reply:
Done in https://github.com/stellar/java-stellar-anchor-sdk/pull/2/commits/75e3e6ffb8cbe1066b5dc46027391ef9c3d2009c